### PR TITLE
refactor: Use Fr type from kilic/bls12-381 for BBS+

### PR DESCRIPTION
**Pull request by kdimak**
Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

#2248

One missing thing is support of `Fr.Inverse()` in https://github.com/kilic/bls12-381. It was temporarily solved at fork https://github.com/trustbloc/bls12-381 with implementation adopted from https://github.com/phoreproject/bls

### CODE REVIEW

**Review by Baha-sk — COMMENTED** [review]
Just wanna confirm, with this change we don't need phorprject anymore @kdimak ?

**Review by Baha-sk — APPROVED** [review]

**Review by llorllale — APPROVED** [review]

File: pkg/doc/bbs/bbs12381g2pub/bbs.go [hunk 1]
```diff
@@ -116,55 +126,58 @@ func (bbs *BBSG2Pub) SignWithKey(messages [][]byte, privKey *PrivateKey) ([]byte
...
 	if err != nil {
 		return nil, fmt.Errorf("compute B point: %w", err)
 	}
 
-	exp := privKey.GetFRElement().Copy()
-	exp.AddAssign(e)
-	sig := b.MulFR(exp.Inverse().ToRepr())
+	exp := bls12381.NewFr().Set(privKey.FR)
+	exp.Add(exp, e)
+	exp = exp.Inverse()
```
Comment (troyronda) [inline pkg/doc/bbs/bbs12381g2pub/bbs.go:146]: @kdimak @llorllale @Baha-sk Looks like the upstream function signature (that was just added) is a bit different:
https://github.com/kilic/bls12-381/blob/a288617c07f1bd60613c43dbde211b4a911e4791/fr.go#L257

**Review by troyronda — APPROVED** [review]

**Review by soluchok — APPROVED** [review]

**Comment by kdimak** [comment]
> Just wanna confirm, with this change we don't need phorprject anymore @kdimak ?

Yes. This dependency is dropped from all `go.mod`-s.

File: pkg/doc/bbs/bbs12381g2pub/bbs.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/bbs/bbs12381g2pub/bbs.go:146]: I will update our fork and use the original kilic function (and double-check in works as expected of course).

**Comment by troyronda** [comment]
@kdimak will create another PR to use the new inverse function.

**Comment by troyronda** [comment]
https://github.com/hyperledger/aries-framework-go/issues/2293

**Pull request merged**